src/docs/arch-block.dox

```diff
@@ -1,9 +1,132 @@
+of page, whether it is compressed, whether it is encrypted, its write generation
+, etc.
+
+The block manager header is simpler. It contains the size of the following disk
+image as well as an optional checksum for when block-level checksums are
+enabled.
+
+@subsection desc_block Descriptor block
+
+As discussed earlier, a file is divided up into blocks. The first block in a file
+is special as it contains metadata about the file and is referred to as the
+"descriptor block". It contains the WiredTiger major and minor version, a
+checksum of the block contents as well as a "magic" number to check against.
+
+The descriptor block serves as a safety check to ensure that the file being
+loaded into the block manager is actually a WiredTiger data file, that it
+belongs to a compatible version of WiredTiger and that the entire file has not
+been corrupted. WiredTiger also offers block-level checksums to defend against
+file corruption at a more granular level, which can be found in the
+\ref checksum section.
+
+@subsection block_lists Extent lists
+
+Internally, WiredTiger uses a data structure called an extent list or a
+\c WT_EXTLIST to track file usage. An extent list consists of a series of
+extents (or \c WT_EXT elements). Each extent tracks a file offset and size to
+track a portion of the file.
+
+There are three extent lists that are maintained per checkpoint:
+
+- \c alloc: The extents allocated for a given checkpoint.
+- \c avail: The extents that are unused and available for allocation.
+- \c discard: The extents freed in the current checkpoint. After the checkpoint
+completes, these extents are shifted to the \c avail extent list.
+
+The alloc and discard extent lists are maintained as a skiplist sorted by file
+offset. The avail extent list also maintains an extra skiplist sorted by the
+extent size to aid with allocating new blocks.
+
+@section configuration Configuration options
+
+There are a number of configuration options that affect the block manager's
+behavior. This does not aim to be an exhaustive list, however, these are the
+configuration options that are more commonly of interest to users.
+
+All of the configuration options below are passed into the \c WT_SESSION::create
+API at the time of file creation.
+
+@subsection allocation_size Allocation size
+
+The \c allocation_size configuration controls the file unit allocation size. Any
+blocks allocated by the block manager must be a multiple of this value.
+
+For example, if we specify an \c allocation_size of \c 4KB, blocks of size \c 8KB
+and \c 12KB would be permitted but NOT \c 10KB. Higher allocation sizes may
+yield better compression rates, yet may cause larger page sizes. The allocation
+size is set to \c 4KB by default which is a good choice unless your OS or storage
+device has special requirements.
+
+@subsection block_compression Block compression
+
+The \c block_compressor configuration controls the compression algorithm used
```

Comment:
> Alex C: Was considering adding a snippet showing the WT_COMPRESSOR struct and the methods required (and the same for WT_ENCRYPTOR below). Do we think that's useful?
>> Keith B: Encryption & compression are almost entirely handled in the upper, Btree layer, not the block manager.
>>> Alex C: Hmm, maybe it should be a separate page then? Encryption and compression were mentioned in the ticket.
>>>> Keith B: I don't know anything about the ticket's goals or history, but I can't think of any code in the block manager that deals with encryption or compression. There's a minor hook that adjusts compression based on the compression state, but that's about it.
>>>>> Alex C: The goal of the project is to write documentation about the system for developers and other people who might be interested such as TSEs. The goal of this ticket is to cover the block manager portion of the system. We're also talking about what data gets passed to the block manager and how WiredTiger uses the API, so I'd say it still fits with this page. The configuration is even referred to as block_compressor, so I'd worry about putting it in a different guide. Maybe clearing up the title to make it clear that it's about more than just the "block manager" component strictly, might help.

Reply:
I'll keep the section on encryption but not expand it much as it's a side topic. @ddanderson could you comment on whether you think that's the right decision?

Reply:
@luke-pearson, (sorry I missed that you tagged me on this).  Keith B's right - encryption and compression is handled in the btree.  I totally misremembered where those changes were made when I put the comments in the ticket.  So let's remove it here, thanks.